test/gatekeeper_controller_test.go

```diff
@@ -62,33 +127,314 @@ var _ = Describe("Gatekeeper", func() {
+												Values: []string{
+													"webhook",
+												},
+											},
+										},
+									},
+									TopologyKey: "kubernetes.io/hostname",
+								},
+							},
+						},
+					},
+				}
+				Expect(webhookDeployment.Spec.Template.Spec.Affinity).To(BeEquivalentTo(defaultWebhookAffinity))
+			})
+
+			By("Checking default node selector", func() {
+				defaultConfig := map[string]string{
+					"kubernetes.io/os": "linux",
+				}
+				Expect(auditDeployment.Spec.Template.Spec.NodeSelector).To(BeEquivalentTo(defaultConfig))
+				Expect(webhookDeployment.Spec.Template.Spec.NodeSelector).To(BeEquivalentTo(defaultConfig))
+			})
+
+			By("Checking default pod annotations", func() {
+				defaultConfig := map[string]string{
+					"container.seccomp.security.alpha.kubernetes.io/manager": "runtime/default",
+				}
+				Expect(auditDeployment.Spec.Template.Annotations).To(BeEquivalentTo(defaultConfig))
+				Expect(webhookDeployment.Spec.Template.Annotations).To(BeEquivalentTo(defaultConfig))
+			})
+
+			By("Checking default tolerations", func() {
+				Expect(auditDeployment.Spec.Template.Spec.Tolerations).To(BeNil())
+				Expect(webhookDeployment.Spec.Template.Spec.Tolerations).To(BeNil())
+			})
+
+			By("Checking default resource limits and requests", func() {
+				defaultConfig := corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("1000m"),
+						corev1.ResourceMemory: resource.MustParse("512Mi"),
+					},
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("100m"),
+						corev1.ResourceMemory: resource.MustParse("256Mi"),
+					},
+				}
+				assertResources(defaultConfig, auditDeployment.Spec.Template.Spec.Containers[0].Resources)
+				assertResources(defaultConfig, webhookDeployment.Spec.Template.Spec.Containers[0].Resources)
+			})
+
+			By("Checking default image", func() {
+				auditImage, auditImagePullPolicy, err := getDefaultImage(auditFile)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(auditDeployment.Spec.Template.Spec.Containers[0].Image).To(Equal(auditImage))
+				Expect(auditDeployment.Spec.Template.Spec.Containers[0].ImagePullPolicy).To(Equal(auditImagePullPolicy))
+				webhookImage, webhookImagePullPolicy, err := getDefaultImage(webhookFile)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(webhookDeployment.Spec.Template.Spec.Containers[0].Image).To(Equal(webhookImage))
+				Expect(webhookDeployment.Spec.Template.Spec.Containers[0].ImagePullPolicy).To(Equal(webhookImagePullPolicy))
+			})
+
+			By("Checking default failure policy", func() {
+				defaultPolicy := admregv1.Ignore
+				validatingWebhookConfiguration := &admregv1.ValidatingWebhookConfiguration{}
+				Eventually(func() error {
+					return K8sClient.Get(ctx, validatingWebhookName, validatingWebhookConfiguration)
+				}, waitTimeout, pollInterval).ShouldNot(HaveOccurred())
+				Expect(auditDeployment).NotTo(BeNil())
```

Comment:
`s/auditDeployment/validatingWebhookConfiguration/` but I don't think this check is necessary as we've just initialized it.